src/net/peer/connect/mod.rs

```diff
@@ -124,9 +124,10 @@ pub fn connect<UID: Uid>(
     name_hash: NameHash,
     our_info: PrivConnectionInfo<UID>,
     their_info: PubConnectionInfo<UID>,
-    config: ConfigFile,
+    config: &ConfigFile,
     peer_rx: UnboundedReceiver<ConnectMessage<UID>>,
 ) -> BoxFuture<Peer<UID>, ConnectError> {
+    let config = config.clone();
```

Comment:
redundant copy?

Reply:
No, `config` is passed to a closure later on.

Reply:
right, I got confused because `connect_directly()` call is in the same scope.
Turns out it's implementation passes config into closure. Wouldn't it be more readable, if  `connect_directly()` took reference, made a copy and passed that copy to a closure?

Reply:
`connect_directly` does take a reference, it's further down at the bottom of the function that `config` gets passed to a closure. Here's it needs to be owned so that the resulting future can be `'static`.

Reply:
See line 161